tracer/test/test-applications/azure-functions/Samples.AzureFunctions.V4Isolated/AllTriggers.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Http.Headers;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Samples.AzureFunctions.AllTriggers;
+
+public class AllTriggers
+{
+    private readonly IHostApplicationLifetime _lifetime;
+    private const string AtMidnightOnFirstJan = "0 0 0 1 Jan *";
+    private static readonly HttpClient HttpClient = new(); 
+    private static readonly ManualResetEventSlim _mutex = new(initialState: false, spinCount: 0);
+
+    private readonly ILogger _logger;
+
+    public AllTriggers(ILoggerFactory loggerFactory, IHostApplicationLifetime lifetime)
+    {
+        _lifetime = lifetime;
+        _logger = loggerFactory.CreateLogger<AllTriggers>();
+    }
+
+    [Function("TriggerAllTimer")]
+    public async Task TriggerAllTimer([TimerTrigger(AtMidnightOnFirstJan, RunOnStartup = true)] TimerInfo myTimer)
+    {
+        _logger.LogInformation($"Profiler attached: {SampleHelpers.IsProfilerAttached()}");
+        _logger.LogInformation($"Profiler assembly location: {SampleHelpers.GetTracerAssemblyLocation()}");
+
+        var envVars = string.Join(", ", SampleHelpers.GetDatadogEnvironmentVariables().Select(x => $"{x.Key}={x.Value}"));
+        _logger.LogInformation("$Profiler env vars: {EnvVars}", envVars);
+
+        _logger.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+        await CallFunctionHttp("trigger");
+        _logger.LogInformation($"Trigger All Timer complete: {DateTime.Now}");
+        _mutex.Set();
+    }
+
+    [Function("ExitApp")]
+    public void ExitApp([TimerTrigger(AtMidnightOnFirstJan, RunOnStartup = true)] TimerInfo myTimer)
+    {
+        _logger.LogInformation($"Waiting for mutex");
+        if (!_mutex.Wait(TimeSpan.FromMinutes(5)))
+        {
+            _logger.LogError($"Error waiting for mutex: not obtained after 5 minutes!");
+        }
+
+        _logger.LogInformation($"Pausing for 3s");
+        Thread.Sleep(3_000); // just need time for the TriggerAllTimer to finish up etc
+
+        _logger.LogInformation($"Calling StopApplication()");
+        _lifetime.StopApplication();
+
+        // brutally kill the host, as can't find any other way to signal it should stop
+        foreach (var process in Process.GetProcessesByName("func"))
+        {
+            _logger.LogInformation("Killing {PID} ({Name}", process.Id, process.ProcessName);
+            process.Kill();
+        }
```

Comment:
Personally, I think no. But it could potentially. Trouble is we don't know _when_ to kill it, which leads to flakiness. Having the app shut itself down resolves that issue, it's just a bit annoying the way it works here 🙄 I'd rather keep this messiness than risk flake tbh

Reply:
> I'd rather keep this messiness than risk flake tbh

Obviously. Though I was raising this because I thought it would be easier to avoid flakiness on the other side. Like we kill everything when we have received all spans or after a timeout. Here waiting 3 seconds made me think of the risk of flakiness.
But if you thought it through, I'm all good with your proposal